canon_module/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+#[cfg(not(feature = "host"))]
```

Comment:
Some comments here would be nice since I don't see the relation with the Repr at all for this code.

Reply:
this is wip at the moment and might not work.